<launch>

  <arg name="target_distance"  default="1.2" />
  <arg name="t_kp"             default="2.580" />
  <arg name="t_kd"             default="0.047" />
  <arg name="r_kp"             default="2.200" />
  <arg name="r_kd"             default="0.100" />

  <arg name="marker_size"          default="9" /> 
  <arg name="max_new_marker_error" default="0.08" />
  <arg name="max_track_error"      default="0.2" />
  <arg name="cam_image_topic"      default="/camera/rgb/image_raw" />
  <arg name="cam_info_topic"       default="/camera/rgb/camera_info" />
  <arg name="output_frame"         default="/camera_rgb_optical_frame" />



  <!--创建二维码世界-->
  <include file="$(find turtlebot_gazebo)/launch/turtlebot_world.launch">
    <arg name="world_file" value="/home/haiden/worlds/marker.world" />
  </include>


  <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
    <param name="marker_size"          type="double" value="$(arg marker_size)" />
    <param name="max_new_marker_error" type="double" value="$(arg max_new_marker_error)" />
    <param name="max_track_error"      type="double" value="$(arg max_track_error)" />
    <param name="output_frame"         type="string" value="$(arg output_frame)" />

    <remap from="camera_image" to="$(arg cam_image_topic)" />
    <remap from="camera_info"  to="$(arg cam_info_topic)" />
  </node>

  <param name="target_distance"   type="double"  value="$(arg target_distance)"  />
  <param name="t_kp"   type="double"  value="$(arg t_kp)"  />
  <param name="t_kd"   type="double"  value="$(arg t_kd)"  />
  <param name="r_kp"   type="double"  value="$(arg r_kp)"  />
  <param name="r_kd"   type="double"  value="$(arg r_kd)"  />

  <node name="ar_location" pkg="ar_location" type="ar_location" output="screen" />

</launch>


